tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc

```diff
@@ -99,4 +108,454 @@ TEST_F( EnergyPlusFixture, ConvectionCoefficientsTest_ConvectionCofficients )
+		"    FLOOR,                    !- Surface Type",
+		"    WALL-1,                  !- Construction Name",
+		"    Zone 1,                  !- Zone Name",
+		"    Outdoors,                !- Outside Boundary Condition",
+		"    ,                        !- Outside Boundary Condition Object",
+		"    SunExposed,              !- Sun Exposure",
+		"    WindExposed,             !- Wind Exposure",
+		"    0.50000,                 !- View Factor to Ground",
+		"    4,                       !- Number of Vertices",
+		"    0.0,0.0,3.0,  !- X,Y,Z ==> Vertex 1 {m}",
+		"    10.0,0.0,3.0,  !- X,Y,Z ==> Vertex 2 {m}",
+		"    10.0,10.0,3.0,  !- X,Y,Z ==> Vertex 3 {m}",
+		"    0.0,10.0,3.0;  !- X,Y,Z ==> Vertex 4 {m}",
+
+		"  BuildingSurface:Detailed,",
+		"    Horizontal Down Floor,                 !- Name",
+		"    FLOOR,                    !- Surface Type",
+		"    WALL-1,                  !- Construction Name",
+		"    Zone 1,                  !- Zone Name",
+		"    Outdoors,                !- Outside Boundary Condition",
+		"    ,                        !- Outside Boundary Condition Object",
+		"    SunExposed,              !- Sun Exposure",
+		"    WindExposed,             !- Wind Exposure",
+		"    0.50000,                 !- View Factor to Ground",
+		"    4,                       !- Number of Vertices",
+		"    0.0,0.0,3.0,  !- X,Y,Z ==> Vertex 1 {m}",
+		"    0.0,10.0,3.0,  !- X,Y,Z ==> Vertex 4 {m}",
+		"    10.0,10.0,3.0,  !- X,Y,Z ==> Vertex 3 {m}",
+		"    10.0,0.0,3.0;  !- X,Y,Z ==> Vertex 2 {m}",
+
+		"  Construction,",
+		"    WALL-1,                  !- Name",
+		"    GP01;                    !- Outside Layer",
+
+		"  Material,",
+		"    GP01,                    !- Name",
+		"    MediumSmooth,            !- Roughness",
+		"    1.2700000E-02,           !- Thickness {m}",
+		"    0.1600000,               !- Conductivity {W/m-K}",
+		"    801.0000,                !- Density {kg/m3}",
+		"    837.0000,                !- Specific Heat {J/kg-K}",
+		"    0.9000000,               !- Thermal Absorptance",
+		"    0.7500000,               !- Solar Absorptance",
+		"    0.7500000;               !- Visible Absorptance",
+
+	} );
+
+	ASSERT_FALSE( process_idf( idf_objects ) );
+
+	bool errorsFound( false );
+	HeatBalanceManager::GetProjectControlData( errorsFound ); // read project control data
+	EXPECT_FALSE( errorsFound ); // expect no errors
+
+	errorsFound = false;
+	HeatBalanceManager::GetMaterialData( errorsFound ); // read material data
+	EXPECT_FALSE( errorsFound ); // expect no errors
+
+	errorsFound = false;
+	HeatBalanceManager::GetConstructData( errorsFound ); // read construction data
+	EXPECT_FALSE( errorsFound ); // expect no errors
+
+	HeatBalanceManager::GetZoneData( errorsFound );
+	ASSERT_FALSE( errorsFound );
+
+	SurfaceGeometry::SetupZoneGeometry( errorsFound );
+	ASSERT_FALSE( errorsFound );
+	HeatBalanceManager::AllocateHeatBalArrays();
+	HeatBalanceSurfaceManager::AllocateSurfaceHeatBalArrays();
```

Comment:
I started feeling like this was a lot of setup needed for this, but it actually makes great sense.  I like this.